colorful/widgets.py

```diff
@@ -2,10 +2,56 @@
 
 import json
 
-from django.forms.widgets import TextInput
+from django.forms.widgets import CheckboxInput, MultiWidget, TextInput
 from django.utils.safestring import mark_safe
 
 
+class NullableColorFieldWidget(MultiWidget):
+    def __init__(self, colors=None, attrs=None):
+        _widgets = [
+            CheckboxInput(attrs={'checked': 'checked', 'style': 'display:none'}),
+            ColorFieldWidget(colors, attrs)
+        ]
+        super(NullableColorFieldWidget, self).__init__(_widgets, attrs)
+
+    def decompress(self, value):
+        if value:
+            return [True, value]
+        return [False, '']
+
+    def value_from_datadict(self, data, files, name):
+        blank = self.widgets[0].value_from_datadict(data, files, name + '_0')
+        color = self.widgets[1].value_from_datadict(data, files, name + '_1')
+
+        if blank:
+            return None
+        else:
+            return color
+
+    @staticmethod
+    def render_script(color_id, checkbox_id):
+        return '''
+                <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.1.1/jquery.min.js"></script>
```

Comment:
I don't think jquery is needed here. This can be rewritten as plain js.
Injecting jquery dependency here is would make very hard to maintain and a pain to use it

Reply:
Hi @yakky,
Agree that jquery dependency is not necessary. I've updated the PR with plain js.

Cheers

Reply:
Nice!